joomla/joomla-cms
# improve mobile checks

Pull Request for Issue mobile checks have multiple conditionals when one is sufficient

### Summary of Changes

only check for resolution in user agent once to determine mobile
add common user agent strings that identify mobile
Fix mobile check to be separate from browser.

### Testing Instructions
code review

(could apply patch and check on a mobile device or emulator that mobile is detected)


### Expected result
mobile is easily and correctly detected.


### Actual result
extraneous checks were in additional locations specific to certain browsers


### Documentation Changes Required
none
